src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -401,132 +415,65 @@ private static ShardFilter getShardFilterAtInitialPosition(InitialPositionInStre
         return shardFilter;
     }
 
+    /**
+     * Determine lease synchronization strategy based on the state of the lease table. If isLeaseTableEmpty is passed
+     * in directly, use that as source of truth. Otherwise, fallback to leaseManager's report of lease table state.
+     * The reason for the isLeaseTableEmpty bypass is to avoid any race conditions that could arise from checking
+     * leaseManager's state multiple times in a single shard sync. The state at the beginning of the shard sync
+     * should be consistent throughout the shard sync operation.
+     *
+     * @param isLeaseTableEmpty nullable Boolean (null if caller has no opinion on current state of lease table)
+     * @param leaseManager holds state of lease table
+     */
+    private LeaseSynchronizer getLeaseSynchronizer(@Nullable Boolean isLeaseTableEmpty,
+                                                   ILeaseManager<KinesisClientLease> leaseManager,
+                                                   Map<String, Shard> shardIdToShardMap,
+                                                   Map<String, Set<String>> shardIdToChildShardIdsMap)
+            throws DependencyException, InvalidStateException, ProvisionedThroughputException {
+
+        if (isLeaseTableEmpty == null) {
```

Comment:
In what case would this value ever be null?

Reply:
See above comment.